src/api/batch-import-export-api.ts

```diff
@@ -0,0 +1,31 @@
+import {PackageExportTransport} from "../interfaces/package-export-transport";
+import {httpClientV2} from "../services/http-client-service.v2";
+import {FatalError} from "../util/logger";
+
+class BatchImportExportApi {
+    public static readonly INSTANCE = new BatchImportExportApi();
+
+    public async findAllActivePackages(flavors: string[], withDependencies: boolean = false): Promise<PackageExportTransport[]> {
+        const queryParams = new URLSearchParams();
+
+        queryParams.set("withDependencies", withDependencies.toString());
```

Comment:
Does the URLSearchParams constructor allow setting these values while constructing the object instead of assigning values? If yes, then it might be a good idea to utilize that instead to try and preserve referential transparency

Reply:
It does have some support, but instead of appending flavors we would have to join the list into a string (`flavors.join(",")` and allow it to be auto-parsed. I would prefer to leave it the way it is to prevent this auto-parsing, which might lead to unexpected behavior.